molecule/source-upgrade/molecule.yml

```diff
@@ -1,7 +1,7 @@
 ---
 dependency:
   name: galaxy
-  role-file: requirements.yml
+  requirements-file: requirements.yml
```

Comment:
I believe those lines do not effect anything there can you try removing them?
In the docs they are found beneath options:
https://molecule.readthedocs.io/en/stable/configuration.html#ansible-galaxy

Reply:
it is being used at dependency test: https://github.com/pulp/pulp_installer/runs/1162604142?check_suite_focus=true#step:9:28

Reply:
the requirements.yml is read, because this is the default value of the dependency class in molecule.
Line 4 of this file is actually ignored, because it _should_ have been:
```
options:
  requirements-file: requirements.yml
```
which will then be interpreted right in modern molecule, but call `ansible-galaxy install -r requirements.yml --requirements-file requirements.yml` on py2's molecule and rightfully fail with "I don't understand that parameter".
It's a mess and it took me some time to see what's going wrong.

Reply:
you are right, `requirements-file` does not work, only works with `roles-file` which does not install the collection

Reply:
And to add to the confusion, on ansible 2.10 is _does_ install the collections in requirements.